kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/controller/DefaultRestControllerImpl.java

```diff
@@ -146,16 +146,21 @@ public KieServerSetup connect(KieServerInfo serverInfo) {
 
         KieServerConfig config = currentState.getConfiguration();
         if (controllers != null && !controllers.isEmpty()) {
+            KieServerSetup kieServerSetup = null;
             for (String controllerUrl : controllers) {
 
                 if (controllerUrl != null && !controllerUrl.isEmpty()) {
-                    KieServerSetup kieServerSetup = connectToSingleController(serverInfo, config, controllerUrl);
-                    if (kieServerSetup != null && kieServerSetup.hasNoErrors()) {
-                        return kieServerSetup;
+                    KieServerSetup kieServerSetupController = connectToSingleController(serverInfo, config, controllerUrl);
```

Comment:
So now the Kie server will connect or disconnect to all controllers?

Reply:
actually it notifies all of them and getthe ke server setup from the first one. with BC cluster (HA) all BC instances needs to be notified that a server is connected.